src/Octoshift/Services/AwsApi.cs

```diff
@@ -34,9 +36,17 @@ private static AmazonS3Client BuildAmazonS3Client(string awsAccessKeyId, string
             AWSConfigsS3.UseSignatureVersion4 = true;
         }
 
-        return awsSessionToken.HasValue()
-            ? new AmazonS3Client(awsAccessKeyId, awsSecretAccessKey, awsSessionToken, regionEndpoint)
-            : new AmazonS3Client(awsAccessKeyId, awsSecretAccessKey, regionEndpoint);
+        var creds = awsSessionToken.HasValue()
+            ? (AWSCredentials)new SessionAWSCredentials(awsAccessKeyId, awsSecretAccessKey, awsSessionToken)
+            : new BasicAWSCredentials(awsAccessKeyId, awsSecretAccessKey);
+
+        var config = new AmazonS3Config
+        {
+            RegionEndpoint = regionEndpoint,
+            Timeout = TimeSpan.FromMinutes(5)
```

Comment:
How does this timeout relate to the CancellationToken timeout? is the timeout going to be the 5 mins from here or the 10 mins from the CTS?

Reply:
I believe the Cancellation Token timeout will apply to the entire upload operation, and in the case of a large file, the Timeout on the AmazonS3Config will apply to the sub-requests of the multi-part upload.